tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <arch/cpu.h>
+#include <arch/arm/cortex_m/cmsis.h>
+
+/* Offset for the Direct interrupt used in this test. */
+#define DIRECT_ISR_OFFSET (CONFIG_NUM_IRQS - 1)
+
+static volatile int test_flag;
+
+void arm_direct_isr_handler_0(void *args)
+{
+	ARG_UNUSED(args);
+
+	test_flag = 1;
+}
+
+void arm_direct_isr_handler_1(void *args)
+{
+	ARG_UNUSED(args);
+
+	test_flag = 2;
+}
+
+void test_arm_dynamic_direct_interrupts(void)
+{
+	int post_flag = 0;
+
+	/* Place the dynamic interrupt dispatcher (with no rescheduling)
+	 * in the ROM ISR table.
+	 */
+	ARM_IRQ_DIRECT_DYNAMIC_CONNECT(DIRECT_ISR_OFFSET, 0, 0, no_reschedule);
+
+	/* Attach the ISR handler at run time. */
+	arch_irq_connect_dynamic(DIRECT_ISR_OFFSET, 0 /* highest priority */,
```

Comment:
should this instead use `irq_connect_dynamic`?

Reply:
done